### PR TITLE
[ISSUE apache#6726] Fix bug DefaultMQProducer#request method invoke t…

### PR DESCRIPTION
**Pull request by haiyanghan**
[ISSUE #6726]

### CODE REVIEW

**Review by tsunghanjacktsai — COMMENTED** [review]
Hey @haiyanghan , thanks for your effort at first. I probably see what you're trying to do here according to the issue you created, but the meaning of message sending in MQ just simply means that the Broker would return a corresponding response once the message arrived without waiting it to be consumed. Of course, you could detail your thought if I've misunderstood your intention for doing this.

**Comment by haiyanghan** [comment]
> Hey @haiyanghan , thanks for your effort at first. I probably see what you're trying to do here according to the issue you created, but the meaning of message sending in MQ just simply means that the Broker would return a corresponding response once the message arrived without waiting it to be consumed. Of course, you could detail your thought if I've misunderstood your intention for doing this.

Yes, when the message arrives at the broker, it returns, which is the logic of the original request method.
But the request method should be based on the request reply mode, waiting for the message to be consumed

**Comment by haiyanghan** [comment]
> Hey @haiyanghan , thanks for your effort at first. I probably see what you're trying to do here according to the issue you created, but the meaning of message sending in MQ just simply means that the Broker would return a corresponding response once the message arrived without waiting it to be consumed. Of course, you could detail your thought if I've misunderstood your intention for doing this.



> Hey @haiyanghan , thanks for your effort at first. I probably see what you're trying to do here according to the issue you created, but the meaning of message sending in MQ just simply means that the Broker would return a corresponding response once the message arrived without waiting it to be consumed. Of course, you could detail your thought if I've misunderstood your intention for doing this.

Sorry, my English proficiency is not very good and the problem description is not very clear

File: broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java [hunk 1]
```diff
@@ -1025,9 +1025,7 @@ public void registerProcessor() {
         replyMessageProcessor.registerSendMessageHook(sendMessageHookList);
 
         this.remotingServer.registerProcessor(RequestCode.SEND_REPLY_MESSAGE, replyMessageProcessor, replyMessageExecutor);
-        this.remotingServer.registerProcessor(RequestCode.SEND_REPLY_MESSAGE_V2, replyMessageProcessor, replyMessageExecutor);
```
Comment (aaron-ai) [inline broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java:1028]: Why is the `SEND_REPLY_MESSAGE_V2` being removed? Will this cause any compatibility issues?

Comment (haiyanghan) [reply broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java:1028]: there may compatibility issues, A better way is to add a new one SEND_REPLY_MESSAGE_V3 message code

**Review by Guiqu1aixi — APPROVED** [review]

**Comment by haiyanghan** [comment]
@Guiqu1aixi  hi, look here

File: broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java (hunk 1 above)
Comment (haiyanghan) [reply broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java]: @aaron-ai hi, I have optimized the compatibility issue, Preserved `SEND_ REPLY_ MESSAGE` and `SEND_ REPLY_ MESSAGE_ V2` request code And the corresponding `processor`

**Pull request closed without merging**